### DISCUSSION TITLE
Adding deleted files to a commit

### DISCUSSION THREAD

**Discussion by MTRNord** [GitoxideLabs/gitoxide#871] [Q&A] [answered]
Hi,

I am coming from the git2 crate. When I wanted to delete a file from the repo I would call `index.remove_path()`. In gitoixide I fail to find how to do this. Basically the equivalent to `git add` where the file was deleted. (Or in even simpler terms -> In commit A I added file A. In commit B I deleted file A. How do I tell the commit that file A has been deleted in commit B?)

**Comment by Byron** [comment, marked as answer]
Index and worktree modifications aren't yet supported, but it will be getting there even though things like `gix add` will take a while to land as they are not currently needed by the top priority, `cargo`.

With that said, I invite you to follow [this tracking issue](https://github.com/Byron/gitoxide/issues/301) which contains a section about *adding* entries to the index.

**Reply by MTRNord** [reply]
Thanks :) 👍
